124-stretch-cluster.md

```diff
@@ -0,0 +1,1795 @@
+
+**Key Finding:** Stretch clusters deployed in same datacenter configurations **maintain full throughput** while providing high availability across independent Kubernetes clusters and availability zones.
+The latency increase is acceptable for most production workloads and provides significant operational benefits through fault domain isolation.
+
+**Testing Methodology Note:** Additional latency sensitivity tests (10ms, 50ms, 100ms) used Chaos Mesh to inject artificial latency between **all pods**, including those within the same cluster.
+This represents a **worst-case scenario**, real world same datacenter deployments only experience latency on cross cluster communication, resulting in significantly better performance than these conservative test results suggest.
+
+#### Use Case Suitability
+
+Stretch Kafka clusters are **optimized for specific deployment scenarios**.
+The following guidance helps users determine when stretch clusters provide value:
+
+##### ✅ Recommended: same datacenter Multi-AZ (Primary Use Case)
+
+**Scenario:** Multiple Kubernetes clusters within the same data center, across different availability zones.
+
+**Network Characteristics:**
+- Inter-cluster latency: < 5ms (validated < 1ms in testing)
+- Packet loss: < 0.1%
+- Jitter: < 2ms
+- Bandwidth: >= 10 Gbps
+
+<Let's add the Graphs here before we submit the proposal>
+
+**Expected Performance:**
+- **Throughput:** 90-100% of single cluster baseline ✅
+- **Latency:** +50-100% overhead (acceptable for HA benefit)
+- **Availability:** Survives entire Kubernetes cluster failure
+
+**Benefits:**
+- **High Availability:** Independent control planes eliminate single point of failure
+- **Fault Isolation:** Failure of one Kubernetes cluster (control plane, etcd, networking) does not impact Kafka
+- **Compliance:** Meets requirements for separate failure domains
+- **Performance:** Excellent - full throughput maintained
+
+**real world Validation:** Testing demonstrated 50,000 msg/sec throughput with 80ms average latency across 3 clusters in same datacenter.
+
+**Example Architecture:**
+```
+Datacenter: US-East-1
+├── Kubernetes Cluster 1 (AZ-1a): 2 controllers, 9 brokers
+├── Kubernetes Cluster 2 (AZ-1b): 2 controllers, 9 brokers
+└── Kubernetes Cluster 3 (AZ-1c): 1 controller, 9 brokers
+
+Network: Low-latency private network (< 1ms)
+Benefit: Survives entire AZ failure or K8s control plane outage
+```
+
+##### Use with Caution: Metro Area
+
+**Scenario:** Kubernetes clusters in nearby datacenters within metro area (< 50km).
+
+**Network Characteristics:**
+- Inter-cluster latency: 5-50ms
+- Requires highly stable, dedicated network links
+
+**Expected Performance:**
+- **Throughput:** 10-30% of baseline (severe degradation)
+- **Latency:** 1-10 seconds average
+- **Stability:** Sensitive to network jitter
+
+**Testing Data:** At 10ms added latency, throughput drops to 17,000 msg/sec (66% reduction). At 50ms, only 4,300 msg/sec remains (91% reduction).
```

Comment:
Why does throughput drop with latency? The general expectation would be stability issues rather than throughput drop. Were stability issues cause of the throughput drop?